setup.py

```diff
@@ -0,0 +1,60 @@
+import subprocess
+import sys
+
+from setuptools import setup
+from setuptools.command.test import test as TestCommand
+
+
+class Tox(TestCommand):
+    user_options = [('tox-args=', 'a', "Arguments to pass to tox")]
+
+    def initialize_options(self):
+        TestCommand.initialize_options(self)
+        self.tox_args = None
+
+    def finalize_options(self):
+        TestCommand.finalize_options(self)
+        self.test_args = []
+        self.test_suite = True
+
+    def run_tests(self):
+        # import here, cause outside the eggs aren't loaded
+        import tox
+        import shlex
+
+        args = self.tox_args
+        if args:
+            args = shlex.split(self.tox_args)
+        errno = tox.cmdline(args=args)
+        sys.exit(errno)
+
+
+def git_version():
+    try:
+        version = subprocess.check_output(
+            'git describe --dirty --always'.split()
+        ).replace('-dirty', '.dev0').strip().decode()
```

Comment:
Might be good to get this to conform to PEP 440, in particular https://www.python.org/dev/peps/pep-0440/#local-version-identifiers. One way to test:

```
$ ./setup.py -h
/usr/lib/python2.7/dist-packages/setuptools/dist.py:294: UserWarning: The version specified ('944732e-dirty') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
  "details." % self.metadata.version
```